crates/sui-indexer/src/apis/indexer_api.rs

```diff
@@ -194,10 +181,13 @@ impl<S: IndexerStore> IndexerApi<S> {
             }
         }?;
 
-        // digests here are of size (limit + 1), where the last one is the cursor for the next page
-        let mut tx_digests = digests_from_db
-            .iter()
-            .map(|digest| {
+        let has_next_page = tx_vec_from_db.len() > limit;
+        tx_vec_from_db.truncate(limit);
+        let next_cursor = tx_vec_from_db
+            .last()
+            .cloned()
```

Comment:
Nit: I think you should be able to avoid the `cloned` here, since you are cloning it on 191 as well

Reply:
removed on L191.